lightly/api/prediction_singletons.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC
+from typing import Optional, List
+
+from lightly.openapi_generated.swagger_client import TaskType
+
+
+class PredictionSingletonRepr(ABC):
+    def __init__(
+        self,
+        type: str,
+        taskName: str,
+        categoryId: int,
+        score: float,
+        cropDatasetId: Optional[str] = None,
+        cropSampleId: Optional[str] = None,
+    ):
+        self.type = type
+        self.taskName = taskName
+        self.categoryId = categoryId
+        self.score = score
+        self.cropDatasetId = cropDatasetId
+        self.cropSampleId = cropSampleId
+
+    def to_dict(self):
+        return {key: value for key, value in vars(self).items() if value is not None}
+
+
+class PredictionSingletonClassificationRepr(PredictionSingletonRepr):
+    def __init__(
+        self,
+        taskName: str,
+        categoryId: int,
+        score: float,
+        probabilities: Optional[List[float]] = None,
+    ):
+        super().__init__(
+            type=TaskType.CLASSIFICATION,
+            taskName=taskName,
+            categoryId=categoryId,
+            score=score,
+        )
+        self.probabilities = probabilities
+
+
+class PredictionSingletonObjectDetectionRepr(PredictionSingletonRepr):
+    def __init__(
+        self,
+        taskName: str,
+        cropDatasetId: str,
```

Comment:
why are the argumentst in a different order order compared to the other classes?

Reply:
I fixed that.